src/App.tsx

```diff
@@ -38,24 +38,53 @@ import { FolderPath } from 'app/drive/types';
 import { manager } from './app/utils/dnd-utils';
 import { AppView } from 'app/core/types';
 import useBeforeUnload from './hooks/useBeforeUnload';
+import PreferencesDialog from 'app/preferences';
 
 interface AppProps {
   isAuthenticated: boolean;
   isInitialized: boolean;
   isFileViewerOpen: boolean;
   isNewsletterDialogOpen: boolean;
   isSurveyDialogOpen: boolean;
+  isPreferencesDialogOpen: boolean;
   fileViewerItem: PreviewFileItem | null;
   user: UserSettings | undefined;
   namePath: FolderPath[];
   dispatch: AppDispatch;
 }
 
 const App = (props: AppProps): JSX.Element => {
+  const {
+    isInitialized,
+    isAuthenticated,
+    isFileViewerOpen,
+    isNewsletterDialogOpen,
+    isSurveyDialogOpen,
+    isPreferencesDialogOpen,
+    fileViewerItem,
+  } = props;
   const token = localStorageService.get('xToken');
   const params = new URLSearchParams(window.location.search);
   const skipSignupIfLoggedIn = params.get('skipSignupIfLoggedIn') === 'true';
   const queryParameters = navigationService.history.location.search;
+  const isOpenPreferencesDialog = params.get('preferences') === 'open';
+  const [haveParamsChanged, setHaveParamsChanged] = useState(false);
+
+  useEffect(() => {
+    window.onpopstate = () => {
+      setHaveParamsChanged(true);
+    };
+  });
+
+  useEffect(() => {
+    if (isOpenPreferencesDialog) {
+      dispatch(uiActions.setIsPreferencesDialogOpen(true));
+      setHaveParamsChanged(false);
+    } else {
+      dispatch(uiActions.setIsPreferencesDialogOpen(false));
+      setHaveParamsChanged(false);
+    }
+  }, [haveParamsChanged]);
```

Comment:
Perhaps it would be a good idea to extract this to group the code that manages the modal state of Preferences

Reply:
Done